### PR TITLE
Add support for MLflow model manage

### LINKED ISSUE #1059: Add support for model management
Model management is an important aspect of the machine learning lifecycle, and currently, the Arena CLI tool lacks support for model management features. [[MLflow](https://github.com/mlflow/mlflow)](https://github.com/mlflow/mlflow) is an open source platform dedicated to managing machine learning lifecycles, with the MLflow Model Registry component acting as a centralized model store, providing a set of APIs, and a UI, to collaboratively manage the full lifecycle of an MLflow Model. It would be meaningful for Arena to support some form of model registry, and MLflow stands out as a good choice. Below are some examples:

- Create a registered model or model version

    ```bash
    # Create a registered model
    $ arena model create \
        --name my-model \
        --tags key1,key2=value2 \
        --description "This is some description about my-model"
    registered model "my-model" created
    
    # Create a model version
    $ arena model create \
        --name my-model \
        --version auto \
        --tags key1,key2=value2 \
        --description "This is some description about my-model v1" \
        --source pvc://my-pvc/models/my-model/1
    model version 1 for "my-model" created
    ```

- Get a registered model or model version

    ```bash
    # Get a registered model
    $ arena model get \
        --name my-model
    Name:                my-model
    LatestVersion        3
    CreationTime:        2024-03-27T19:59:14+08:00
    LastUpdatedTime:     2024-03-27T20:01:35+08:00
    Description:
      This is some description about my-model
    Tags:
      createdBy: arena
      key1: 
      key2: value2
    Versions:
      Version    Source
      ---        ---
      1          pvc://my-pvc/models/my-model/1
      2          pvc://my-pvc/models/my-model/2
      3          pvc://my-pvc/models/my-model/3
    
    # Get a model version
    $ arena model get \
        --name my-model \
        --version 1
    Name:                my-model
    Version:             1
    CreationTime:        2024-03-27T20:01:31+08:00
    LastUpdatedTime:     2024-03-27T20:01:31+08:00
    Source:              pvc://my-pvc/models/my-model/1
    Description:
      This is some description about my-model v1
    Tags:
      createdBy: arena
      key1: 
      key2: value2
    ```

- List all registered models

    ```bash
    $ arena model list    
    NAME                 LATEST_VERSION       LAST_UPDATED_TIME                                                               
    model-1              1                    2024-03-27 20:20:59                                                             
    model-2              2                    2024-03-27 20:21:04                                                             
    my-model             3                    2024-03-27 20:20:45
    ```

- Update a registered model or model version

    ```bash
    # Update a registered model
    $ arena model update \
        --name my-model \
        --description "This is some updated description" \
        --tags key1,key2=value2
    registered model "my-model" updated
    
    # Update a model version
    $ arena model update \
        --name my-model \
        --version 1 \
        --description "This is some updated description" \
        --tags key1,key2=value2
    model version "my-model/1" updated
    
    # Update a model version(delete tags)
    $ arena model update \
        --name my-model \
        --version 1 \
        --tags key1-,key2=value2-
    model version "my-model/1" updated
    ```

- Delete a registered model or model version

    ```bash
    # Delete a registered model with confirmation
    $ arena model delete \
      --name my-model
    Delete a registered model will cascade delete all its model versions. Are you sure you want to perform this operation? (yes/no)
    yes
    registered model "my-model" deleted
    
    # Delete a registered model forcely
    $ arena model delete \
      --name my-model \
      --force
    registered model "my-model" deleted
    
    # Delete a model version with confirmation
    $ arena model delete \
        --name my-model \
        --version 1
    Are you sure you want to perform this operation? (yes/no)
    yes
    model version "my-model/1" deleted
    
    # Delete a model version forcely
    $ arena model delete \
        --name my-model \
        --version 1 \
        --force
    model version "my-model/1" deleted
    ```

### PR DESCRIPTION
**Pull request by ChenYi015**
**What this PR does / why we need it**:
Model management is an important aspect of the machine learning lifecycle, and currently, the Arena CLI tool lacks support for model management features. [MLflow](https://github.com/mlflow/mlflow) is an open source platform dedicated to managing machine learning lifecycles, with the MLflow Model Registry component acting as a centralized model store, providing a set of APIs, and a UI, to collaboratively manage the full lifecycle of an MLflow Model. It would be meaningful for Arena to support some form of model registry, and MLflow stands out as a good choice.

**Which issue(s) this PR fixes**:
Fixes #1059

### CODE REVIEW

File: pkg/commands/model/delete.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		Use:   "delete",
+		Short: "Delete a registered model or model version",
+		Args:  cobra.NoArgs,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			_ = viper.BindPFlags(cmd.Flags())
+
+			if mlflowTrackingUri == "" {
+				return fmt.Errorf("MLflow tracking server URI must be specified by MLFLOW_TRACKING_URI environment variable")
+			}
+			if name == "" {
```
Comment (Syulin7) [inline pkg/commands/model/delete.go:26]: use command.MarkFlagRequired

**Comment by Syulin7** [comment]
/lgtm

@ChenYi015  Thank you for your outstanding contribution！

File: pkg/apis/arenaclient/serving_client.go [hunk 2]
```diff
@@ -82,7 +83,30 @@ func (t *ServingJobClient) GetAndPrint(jobName, version string, jobType types.Se
...
+		}
+	}
+	if mv == nil {
+		mv = &types.ModelVersion{
+			Name:    modelName,
+			Version: modelVersion,
+		}
+	}
+
+	serving.PrintServingJob(job, mv, utils.TransferPrintFormat(format))
```
Comment (cheyang) [inline pkg/apis/arenaclient/serving_client.go:109]: If there is no mlflow installed, what will be shown in CLI mode ?

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**